compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt

```diff
@@ -0,0 +1,144 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:OptIn(ExperimentalMaterial3AdaptiveApi::class)
+
+package com.example.compose.snippets.adaptivelayouts
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.layout.safeContentPadding
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.material3.adaptive.ExperimentalMaterial3AdaptiveApi
+import androidx.compose.material3.adaptive.layout.AnimatedPane
+import androidx.compose.material3.adaptive.layout.PaneAdaptedValue
+import androidx.compose.material3.adaptive.layout.SupportingPaneScaffold
+import androidx.compose.material3.adaptive.layout.SupportingPaneScaffoldRole
+import androidx.compose.material3.adaptive.layout.ThreePaneScaffoldRole
+import androidx.compose.material3.adaptive.layout.ThreePaneScaffoldScope
+import androidx.compose.material3.adaptive.navigation.rememberSupportingPaneScaffoldNavigator
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+
+@Composable
+fun SampleSupportingPaneScaffoldParts() {
+    // [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_nav_and_back]
+    val navigator = rememberSupportingPaneScaffoldNavigator()
+
+    BackHandler(navigator.canNavigateBack()) {
+        navigator.navigateBack()
+    }
+    // [END android_compose_adaptivelayouts_sample_supporting_pane_scaffold_nav_and_back]
+
+    // [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_params]
+    SupportingPaneScaffold(
+        directive = navigator.scaffoldDirective,
+        value = navigator.scaffoldValue,
+        mainPane = { /*...*/ },
+        supportingPane = { /*...*/ },
+    )
+    // [END android_compose_adaptivelayouts_sample_supporting_pane_scaffold_params]
+}
+
+@Composable
+fun SampleSupportingPaneScaffoldFull() {
+    // [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_full]
+    val navigator = rememberSupportingPaneScaffoldNavigator()
+
+    BackHandler(navigator.canNavigateBack()) {
+        navigator.navigateBack()
+    }
+
+    SupportingPaneScaffold(
+        directive = navigator.scaffoldDirective,
+        value = navigator.scaffoldValue,
+        mainPane = {
+            AnimatedPane(modifier = Modifier.safeContentPadding()) {
+                // Main pane content
+                if (navigator.scaffoldValue.secondary == PaneAdaptedValue.Hidden) {
```

Comment:
The `get()` overload here looks a bit nicer in my eyes:

```suggestion
                if (navigator.scaffoldValue[SupportingPaneScaffoldRole.Supporting] == PaneAdaptedValue.Hidden) {
```